.github/workflows/trivy.yml

```diff
@@ -1,11 +1,21 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2024 Intel Corporation.
+
 name: Trivy scan
 
 on:
   workflow_dispatch:
+  schedule:
+    - cron: '45 8 * * *'
   push:
     branches: ["master", "release/**"]
   pull_request:
     branches: ["master", "release/**"]
+    paths:
+      - '**/go.mod'
+      - '**/pom.xml'
+      - '**/requirements.txt'
+      - '**/*trivy*'
```

Comment:
What happens if something new is added to the repo that Trivy should be scanning but of course the developer adding that something new has no idea that Trivy should be scanning it?

Reply:
> What happens if something new is added to the repo that Trivy should be scanning but of course the developer adding that something new has no idea that Trivy should be scanning it?

There is nothing like "something new" in Trivy. Every scan follows the Trivy configuration file, which defines precisely what shall be scanned. We need to modify the Trivy configuration to extend the scope of scans. But it is not a developer's task but rather the security engineer's one—to decide that additional checks are required. So far, Trivy scans Java, Python, and Go dependencies, and this is the basement for file path filtering.

Reply:
I don't think I was very clear in my question.

The problem I see is that the above is only running Trivy scan on PRs that modify one of the listed paths.

But let's say a developer adds some new code in a new path to the source tree and that code should be scanned by Trivy.  But the average developer is not going to know about this Trivy scan workflow and is not going to think about the need to add their path to the list of paths that trigger Trivy scan.

So they get their PR finished and landed without any Trivy scanning but the moment it's landed Trivy scan runs and finds the new code and finds a failure in it.  Now master has to be closed and somebody has a high-priority task to resolve this new scan failure, interrupting their scheduled work and nobody else can get anything landed until the issue is resolved.

All of this could have been prevented by always scanning in PRs, not only when some whitelist of paths are modified.

Reply:
The latest commit limits the list of files to be monitored by the `push` event.

The file list has been extended by all supported by Trivy files - https://aquasecurity.github.io/trivy/v0.56/docs/coverage/language/#supported-languages

The solution does not block any PR from being landed if not Trivy-related.

`schedule` run reports detected issues to the Security tab, which should be somehow monitored.
(We have to define a procedure if no one monitors Security related issues reported via Security tab)

Reply:
> > the Security tab, which should be somehow monitored. (We have to define a procedure if no one monitors Security related issues reported via Security tab)
> 
> This needs to be defined and put in place before we can land anything that **only** reports there.

We already have similar situation with Scorecard where all issues are reported only to Security tab.

Reply:
> We already have similar situation with Scorecard where all issues are reported only to Security tab.

Scorecard issues show up in the PR that introduces them.